src/Telegram.php

```diff
@@ -310,17 +318,18 @@ public function getLastCommandResponse()
      *
      * @param int|null $limit
      * @param int|null $timeout
+     * @param bool $no_database
      *
      * @return \Longman\TelegramBot\Entities\ServerResponse
      * @throws \Longman\TelegramBot\Exception\TelegramException
      */
-    public function handleGetUpdates($limit = null, $timeout = null)
+    public function handleGetUpdates($limit = null, $timeout = null, $no_database = false)
```

Comment:
Instead of adding a boolean parameter, it's better practice to implement this differently.
(Have a look here: https://softwareengineering.stackexchange.com/a/212100)

We could either do this with a `handleGetUpdatesWithoutDatabase($limit = null, $timeout = null)` method, or an extra property that can be set using a `useGetUpdatesWithoutDatabase(bool $handle_get_updates_without_database = true)` method.

The user obviously needs to be aware of the fact that it only works in a loop then!

---

Other option would be to just get the initial updates (1st request in the loop) to get the update ID to handle it, then remember the ID from there ([similar to this](https://github.com/irazasyed/telegram-bot-sdk/blob/ab2321154f124c1b9d979a2087db7aa0561fe9a8/src/Api.php#L1326)). This would allow `handleGetUpdates` to work even outside a loop, but it will send off an extra request to confirm that the updates have been handled each time.
But that is then up to the user.
If we go this route, then we don't need the user to explicitly say if the database should be used, but we'd use it if it's available.

What do you think?

Reply:
Actually that extra request to mark them as read will only run once when in loop because after first loop we will already know the last update id so we can process new and mark previous updates at the same time!

I also noticed a problem here, if it's running in a loop and script is killed previous updates aren't marked as handled, so the next time script runs it reprocesses same updates again! So definitely running without DB can't be a default here. (the whole problem can be handled with signal handler)